bodhi/tests/server/test_models.py

```diff
@@ -626,6 +626,81 @@ def test_days_to_stable_not_meets_testing_requirements_with_date_testing(self):
 
         eq_(update.days_to_stable, 3)
 
+    @mock.patch.dict('bodhi.server.config.config', {'ci.required': True})
+    def test_ci_failed_no_testing_requirements(self):
+        """
+        The Update.meets_testing_requirements() should return False if the
+        builds of an update did not pass CI.
+        """
+        update = self.obj
+        update.autokarma = False
+        update.stable_karma = 1
+        update.builds[0].ci_status = CiStatus.failed
+        update.comment(self.db, u'I found $100 after applying this update.', karma=1,
+                       author=u'bowlofeggs')
+        # Assert that our preconditions from the docblock are correct.
+        eq_(update.meets_testing_requirements, False)
+
+    @mock.patch.dict('bodhi.server.config.config', {'ci.required': True})
+    def test_ci_queued_no_testing_requirements(self):
+        """
+        The Update.meets_testing_requirements() should return False if the
+        builds of an update did not pass CI.
+        """
+        update = self.obj
+        update.autokarma = False
+        update.stable_karma = 1
+        update.builds[0].ci_status = CiStatus.queued
+        update.comment(self.db, u'I found $100 after applying this update.', karma=1,
+                       author=u'bowlofeggs')
+        # Assert that our preconditions from the docblock are correct.
+        eq_(update.meets_testing_requirements, False)
+
+    @mock.patch.dict('bodhi.server.config.config', {'ci.required': True})
+    def test_ci_running_no_testing_requirements(self):
+        """
+        The Update.meets_testing_requirements() should return False if the
+        builds of an update did not pass CI.
+        """
+        self.config['ci.required'] = True
+        update = self.obj
+        update.autokarma = False
+        update.stable_karma = 1
+        update.builds[0].ci_status = CiStatus.running
+        update.comment(self.db, u'I found $100 after applying this update.', karma=1,
+                       author=u'bowlofeggs')
+        # Assert that our preconditions from the docblock are correct.
+        eq_(update.meets_testing_requirements, False)
+
+    @mock.patch.dict('bodhi.server.config.config', {'ci.required': True})
+    def test_ci_missing_no_testing_requirements(self):
+        """
+        The Update.meets_testing_requirements() should return False if the
+        builds of an update did not pass CI.
+        """
+        update = self.obj
+        update.autokarma = False
+        update.stable_karma = 1
+        update.builds[0].ci_status = None
+        update.comment(self.db, u'I found $100 after applying this update.', karma=1,
+                       author=u'bowlofeggs')
+        # Assert that our preconditions from the docblock are correct.
+        eq_(update.meets_testing_requirements, False)
+
+    def test_ci_ci_required_off(self):
```

Comment:
It could be useful to mock the setting being set to ```"False"``` here to test that the bool is properly parsed.